File: backend/app/schemas/legal.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class LegalQARequest(BaseModel):
    question: str
    jurisdiction: str = "federal"


class LegalQAResponse(BaseModel):
    question: str
    answer: str
    jurisdiction: str
    sources: list[str]
    cached: bool


class LegalQAHistoryItem(BaseModel):
    id: UUID
    question: str
    answer: str
    jurisdiction: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/legal.ts
export interface LegalQARequest {
  question: string;
  jurisdiction: string;
}

export interface LegalQAResponse {
  question: string;
  answer: string;
  jurisdiction: string;
  sources: string[];
  cached: boolean;
}

export interface QAHistoryItem {
  id: string;
  question: string;
  answer: string;
  jurisdiction: string;
  created_at: string;
}
